apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4PatientResourceProvider.java

```diff
@@ -906,22 +907,31 @@ private Patient queryDatabaseByHash(
           benesByHashOrIdQueryNanoSeconds,
           matchingBenes.size());
     }
-
     // Then, if we found more than one distinct BENE_ID, or none, throw an error.
     long distinctBeneIds =
         matchingBenes.stream()
             .map(Beneficiary::getBeneficiaryId)
             .filter(Objects::nonNull)
             .distinct()
             .count();
```

Comment:
we should simplify this, we don't need to have two lists here

```java
List<Long> distinctBeneIds =
        matchingBenes.stream()
            .map(Beneficiary::getBeneficiaryId)
            .distinct()
            .collect(Collectors.toList());

    if (distinctBeneIds.size() == 0) {
      throw new NoResultException();
    } else if (distinctBeneIdsList.size() > 1) {
       BfdMDC.put(
          "database_query_by_hash_collision_distinct_bene_ids", Long.toString(distinctBeneIds));
      throw new ResourceNotFoundException(
          "By hash query found more than one distinct BENE_ID: "
// Guessing you meant to put the size here and not the list twice
              + distinctBeneIds.size()
              + ", DistinctBeneIdsList: "
              + distinctBeneIdsList);
   }

      Beneficiary beneficiary = matchingBenes.get(0);
```

Not sure if `.filter(Objects::nonNull)` is relevant when it's placed after `.map(Beneficiary::getBeneficiaryId)`. Would drop the sorting as well, personally, for performance reasons

Reply:
Removed the .filter(Objects::nonNull)